### PR TITLE
Add minitest comparison notes?

### PR DESCRIPTION
**Pull request by andyw8**
Since Ruby ships with two testing frameworks, I think it's important to give some guidance about which to choose, e.g.

* Is `test-unit` a viable choice for new projects, or it just for legacy projects? 
* Is `test-unit` opinionated in ways that differ from minitest?

The description here is incomplete because I don't know the answers, but I'd like to find out.

### CODE REVIEW

**Comment by kou** [comment]
>     * Is `test-unit` a viable choice for new projects

Yes.

> or it just for legacy projects?

No.

>     * Is `test-unit` opinionated in ways that differ from minitest?

test-unit never introduce backward incompatible API changes not like RSpec 3 and minitest 5. If there are backward incompatible changes, they are bugs.

test-unit also supports old (EOL-ed) Ruby as much as possible.

If a testing framework introduce some backward incompatibilities, users can't update their dependencies including testing framework, Ruby and other gems with confidence. Updating multiple dependencies that have backward incompatibilities at once is a heavy task.

**Comment by andyw8** [comment]
Thanks @kou, I've made an update to reflect those points. I realize this is still probably a gross simplification of history and direction of the test-unit, but I hope it's useful as a starting point.

File: README.md [hunk 1]
```diff
@@ -18,6 +18,14 @@ writing tests, checking results and automated testing in Ruby.
 test-unit ships as part of Ruby as a bundled gem. To check which version is included, see https://stdgems.org/test-unit/.
 It's only necessary to install the gem if you need a newer version.
 
+test-unit is one of two testing unit libraries bundled with Ruby.
+The other is [minitest](https://github.com/seattlerb/minitest), which is newer, but test-unit is still active.
```
Comment (kou) [inline README.md:22]: Should we mention that minitest was started after test-unit explicitly? Is it important for users?

I feel that this wording is negative for test-unit.,,

Comment (andyw8) [reply README.md:22]: Yes, I see, how about:

```suggestion
test-unit is the original Ruby unit testing library, and is still active.
It is one of two testing unit libraries bundled with Ruby, the other being [minitest] (https://github.com/seattlerb/minitest),
```

Comment (kou) [reply README.md:22]: It's better than before. Thanks.

"testing unit" -> "unit testing"

Comment (andyw8) [reply README.md]: @kou thanks, updated.

**Comment by kou** [comment]
Thanks.

**Pull request merged**